tsdb/wlog/wlog_test.go

```diff
@@ -208,6 +239,201 @@ func TestWALRepair_ReadingError(t *testing.T) {
 	}
 }
 
+// TestWALIgnoreCorruption ensures that if ignoreCorruption is set, reading will continue on remain valid records.
+func TestWALIgnoreCorruption(t *testing.T) {
+	for name, test := range map[string]struct {
+		corrSgm        int              // Which segment to corrupt.
+		corrFunc       func(f *os.File) // Func that applies the corruption.
+		intactRecs     int              // Total expected records left after the repair.
```

Comment:
This field overlaps with `len(intactRecIndices)`, so it's pointless (forces you to state the same expectation twice per test case).
```suggestion
```